File: Dungeon generation/Dungeon generation/Generator/Generator.cs
using System;
using System.Data;
using Dungeon_generation.Models;
using Dungeon_generation.Utils;

namespace Dungeon_generation.Generator {
  /// <summary>
  /// A class responsible for each step of the dungeon generation.
  /// </summary>
  public class Generator : IGenerator {
    /// <summary>
    /// Initial step: Generate rooms of variable width and height inside a boundary.
    /// </summary>
    public List<Room> generateRooms(
      int attempts,
      int containerWidth,
      int containerHeight,
      int minRoomWidth,
      int maxRoomWidth,
      int minRoomHeight,
      int maxRoomHeight
    ) {
      List<Room> rooms = new List<Room>();
      for (int i = 0; i < attempts; i++) {
        Room room = new Room(
          Helpers.randInt(minRoomWidth, maxRoomWidth),
          Helpers.randInt(minRoomHeight, maxRoomHeight),
          new Coordinate(
            Helpers.randInt(0, containerWidth),
            Helpers.randInt(0, containerHeight)
          )
        );
        // To prevent infinite loops overlapping rooms are thrown away
        if (!rooms.Any(r => Helpers.roomsOverlap(room, r))) {
          rooms.Add(room);
        }
      }
      return rooms;
    }
  }
}

File: visualization/src/services/generator.js
import axios from "axios";
import constants from "../utils/constants";

class Generator {
  async generateRooms(
    attempts,
    containerWidth,
    containerHeight,
    minRoomWidth,
    maxRoomWidth,
    minRoomHeight,
    maxRoomHeight
  ) {
    const { data } = await axios.get(`${constants.API_URL}/generate-rooms`, {
      params: {
        attempts,
        containerWidth,
        containerHeight,
        minRoomWidth,
        maxRoomWidth,
        minRoomHeight,
        maxRoomHeight
      }
    });
    return data;
  }
}

export default Generator;
